azure_pipelines/pr-validation.yml

```diff
@@ -84,6 +84,55 @@ jobs:
       failTaskOnFailedTests: true
       testRunTitle: 'Test Run - $(target)'
 
+- job: fetch_pr_labels
+  displayName: 'Check for PR Label'
+  timeoutInMinutes: 5
+  pool:
+    vmImage: 'macOS-14'
+  steps:
+    - script: |
+        url="https://api.github.com/repos/$BUILD_REPOSITORY_ID/issues/$SYSTEM_PULLREQUEST_PULLREQUESTNUMBER/labels"
+
+        echo "Fetching labels from $url "
+
+        # Temporary file to store the raw response
+        temp_file=$(mktemp)
+
+        # Store the response code and the raw response in separate variables
+        response_code=$(curl -s -w "%{http_code}" -o "$temp_file" "$url")
```

Comment:
Perhaps you can add pipeline access token in the AUTHORIZATION header here.

Reply:
Yes, we will do it as part of [the next quarter's task](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2274#discussion_r1731160243). Thank you.